lib/db.js

```diff
@@ -51,6 +52,7 @@ class LimitDBRedis extends EventEmitter {
     this.configurateBuckets(config.buckets);
     this.prefix = config.prefix;
     this.globalTTL = (config.globalTTL || ms('7d')) / 1000;
+    this.callCounts = new LRU({ max: 50 });
```

Comment:
Can we lost a key count in case of reaching the maximum size of the cache? Shouldn't we handle the 'dispose` function to report those counts to redis?

Reply:
Using a `maxCallCountKey` I tried to test when this _dispose_ and _afterdispose_ will emit.
```
    this.callCounts = new LRU({
      max: config.maxCallCountKey || 50,
      afterDispose: (value, key, reason) => {
        console.log(value);
        this.emit('cache_evict', { value, key, reason })
      }
    });

```
If you look in the following tests; I hit take with different keys (100) but it never call dispose. 
 
```    it.only('should emit cache_evict', (done) => {
      db = new LimitDB({ uri: 'localhost', buckets, prefix: 'tests:', maxCallCountKey: 1 });
      const takeParams = (i) => ({ type: 'global',  key: `skipit${i}`});

      db.on('ready', () => {
        async.each(_.range(0, 100), (i, f) => {
          db.take(takeParams(i), f);
        }, (err) => {
          if (err) return done(err);
          done();
        });
      })
    });
```

Do you think that's ok continue without this and add a task to handle this in the future?

Reply:
Weird, this should be emitted. 
OK with continue and add a task for reviewing this. Could you add the Jira ticket to the code?
Should we run this investigation soon?